src/js/controller/controller.js

```diff
@@ -977,8 +977,20 @@ Object.assign(Controller.prototype, {
             _programController.controls = mode;
         };
 
+        this.addCues = function (cues) {
+            const existingCues = this.getCues();
+            const newCues = existingCues.concat(cues);
+            this.setCues(newCues);
+        };
+
         this.setCues = function (cues) {
-            _model.set('cues', cues);
+            if (Array.isArray(cues)) {
+                _model.set('cues', cues);
+            }
+        };
+
+        this.getCues = function() {
```

Comment:
You don't need this method because the api doesn't use it, and you can just use the model `_model.get('cues');` in `addCues` above.